src/parser/parseTransformAttribute.ts

```diff
@@ -1,105 +1,50 @@
-//@ts-nocheck
 import { iMatrix } from '../constants';
-import { commaWsp, reNum } from './constants';
+import { reNum } from './constants';
 import { multiplyTransformMatrices } from '../util/misc/matrix';
 import { degreesToRadians } from '../util/misc/radiansDegreesConversion';
 import { rotateMatrix } from './rotateMatrix';
 import { scaleMatrix } from './scaleMatrix';
 import { skewMatrix } from './skewMatrix';
 import { translateMatrix } from './translateMatrix';
+import { TMat2D } from '../typedefs';
 
 // == begin transform regexp
-const number = reNum,
-  skewX = '(?:(skewX)\\s*\\(\\s*(' + number + ')\\s*\\))',
-  skewY = '(?:(skewY)\\s*\\(\\s*(' + number + ')\\s*\\))',
-  rotate =
-    '(?:(rotate)\\s*\\(\\s*(' +
-    number +
-    ')(?:' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    '))?\\s*\\))',
-  scale =
-    '(?:(scale)\\s*\\(\\s*(' +
-    number +
-    ')(?:' +
-    commaWsp +
-    '(' +
-    number +
-    '))?\\s*\\))',
-  translate =
-    '(?:(translate)\\s*\\(\\s*(' +
-    number +
-    ')(?:' +
-    commaWsp +
-    '(' +
-    number +
-    '))?\\s*\\))',
-  matrix =
-    '(?:(matrix)\\s*\\(\\s*' +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    commaWsp +
-    '(' +
-    number +
-    ')' +
-    '\\s*\\))',
-  transform =
-    '(?:' +
-    matrix +
-    '|' +
-    translate +
-    '|' +
-    scale +
-    '|' +
-    rotate +
-    '|' +
-    skewX +
-    '|' +
-    skewY +
-    ')',
-  transforms =
-    '(?:' + transform + '(?:' + commaWsp + '*' + transform + ')*' + ')',
-  transformList = '^\\s*(?:' + transforms + '?)\\s*$',
-  // http://www.w3.org/TR/SVG/coords.html#TransformAttribute
-  reTransformList = new RegExp(transformList),
-  // == end transform regexp
-  reTransform = new RegExp(transform, 'g');
+const p = String.raw`(${reNum})`;
+const skewX = String.raw`(skewX)\(${p}\)`;
+const skewY = String.raw`(skewY)\(${p}\)`;
+const rotate = String.raw`(rotate)\(${p}(?: ${p} ${p})?\)`;
+const scale = String.raw`(scale)\(${p}(?: ${p})?\)`;
+const translate = String.raw`(translate)\(${p}(?: ${p})?\)`;
+const matrix = String.raw`(matrix)\(${p} ${p} ${p} ${p} ${p} ${p}\)`;
+const transform = String.raw`(?:${matrix}|${translate}|${rotate}|${scale}|${skewX}|${skewY})`;
+const transforms = String.raw`(?:${transform}*)`;
+const transformList = String.raw`^\s*(?:${transforms}?)\s*$`;
```

Comment:
please don't use String.raw where is no need for it. 1( not sure where here )
Also to ease review, can you explain how you get rid of the commaWsp?

Reply:
Similar to my other work on the path parser, I clean the string in the parse function to remove commas and arbitrary whitespace.